File: symfony-app/src/Entity/Medicament.php
<?php

namespace App\Entity;

use App\Repository\MedicamentRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: MedicamentRepository::class)]
#[ORM\Table(name: 'medicaments')]
class Medicament
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(type: 'string', length: 150)]
    private string $nom;

    #[ORM\Column(type: 'string', length: 50, nullable: true)]
    private ?string $type = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(type: 'string', length: 50, nullable: true)]
    private ?string $forme = null;

    #[ORM\Column(type: 'string', length: 50, nullable: true)]
    private ?string $dosage = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2, nullable: true)]
    private ?string $prix = null;

    #[ORM\Column(type: 'integer', options: ['default' => 0])]
    private int $stock = 0;

    #[ORM\Column(type: 'string', length: 100, nullable: true)]
    private ?string $laboratoire = null;

    #[ORM\Column(type: 'string', length: 120, nullable: true)]
    private ?string $codeBarre = null;

    #[ORM\Column(type: 'datetime', options: ['default' => 'CURRENT_TIMESTAMP'])]
    private \DateTimeInterface $createdAt;

    #[ORM\Column(type: 'datetime', options: ['default' => 'CURRENT_TIMESTAMP'])]
    private \DateTimeInterface $updatedAt;

    #[ORM\OneToMany(mappedBy: 'medicament', targetEntity: StockPharmacy::class, cascade: ['persist', 'remove'])]
    private Collection $stockPharmacies;

    #[ORM\OneToMany(mappedBy: 'medicament', targetEntity: ReponseMedicament::class)]
    private Collection $reponsesMedicaments;


    public function __construct()
    {
        $this->createdAt = new \DateTime();
        $this->updatedAt = new \DateTime();
        $this->stockPharmacies = new ArrayCollection();
        $this->reponsesMedicaments = new ArrayCollection();
    }

    public function getId(): ?int { return $this->id; }
    public function getNom(): string { return $this->nom; }
    public function setNom(string $nom): void { $this->nom = $nom; }
    public function getType(): ?string { return $this->type; }
    public function setType(?string $type): void { $this->type = $type; }
    public function getDescription(): ?string { return $this->description; }
    public function setDescription(?string $description): void { $this->description = $description; }
    public function getForme(): ?string { return $this->forme; }
    public function setForme(?string $forme): void { $this->forme = $forme; }
    public function getDosage(): ?string { return $this->dosage; }
    public function setDosage(?string $dosage): void { $this->dosage = $dosage; }
    public function getPrix(): ?string { return $this->prix; }
    public function setPrix(?string $prix): void { $this->prix = $prix; }
    public function getStock(): int { return $this->stock; }
    public function setStock(int $stock): void { $this->stock = $stock; }
    public function getLaboratoire(): ?string { return $this->laboratoire; }
    public function setLaboratoire(?string $laboratoire): void { $this->laboratoire = $laboratoire; }
    public function getCodeBarre(): ?string { return $this->codeBarre; }
    public function setCodeBarre(?string $codeBarre): void { $this->codeBarre = $codeBarre; }
    public function getCreatedAt(): \DateTimeInterface { return $this->createdAt; }
    public function setCreatedAt(\DateTimeInterface $createdAt): void { $this->createdAt = $createdAt; }
    public function getUpdatedAt(): \DateTimeInterface { return $this->updatedAt; }
    public function setUpdatedAt(\DateTimeInterface $updatedAt): void { $this->updatedAt = $updatedAt; }
    public function getStockPharmacies(): Collection { return $this->stockPharmacies; }
    public function getReponsesMedicaments(): Collection { return $this->reponsesMedicaments; }
}

File: javafx-app/src/main/java/com/santea/model/Medicament.java
package com.santea.model;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Medicament {
    private Integer id;
    private String nom;
    private String type;
    private String description;
    private String forme;
    private String dosage;
    private String prix;
    private Integer stock;
    private String laboratoire;
    private String codeBarre;
    private String imageName;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<Object> stockPharmacies;
    private List<Object> reponsesMedicaments;

    public Medicament() {
        this.stockPharmacies = new ArrayList<>();
        this.reponsesMedicaments = new ArrayList<>();
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getForme() {
        return forme;
    }

    public void setForme(String forme) {
        this.forme = forme;
    }

    public String getDosage() {
        return dosage;
    }

    public void setDosage(String dosage) {
        this.dosage = dosage;
    }

    public String getPrix() {
        return prix;
    }

    public void setPrix(String prix) {
        this.prix = prix;
    }

    public Integer getStock() {
        return stock;
    }

    public void setStock(Integer stock) {
        this.stock = stock;
    }

    public String getLaboratoire() {
        return laboratoire;
    }

    public void setLaboratoire(String laboratoire) {
        this.laboratoire = laboratoire;
    }

    public String getCodeBarre() {
        return codeBarre;
    }

    public void setCodeBarre(String codeBarre) {
        this.codeBarre = codeBarre;
    }

    public String getImageName() {
        return imageName;
    }

    public void setImageName(String imageName) {
        this.imageName = imageName;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    public List<Object> getStockPharmacies() {
        return stockPharmacies;
    }

    public void setStockPharmacies(List<Object> stockPharmacies) {
        this.stockPharmacies = stockPharmacies;
    }

    public List<Object> getReponsesMedicaments() {
        return reponsesMedicaments;
    }

    public void setReponsesMedicaments(List<Object> reponsesMedicaments) {
        this.reponsesMedicaments = reponsesMedicaments;
    }
}
